demo/03-create-cluster.sh

```diff
@@ -99,6 +101,93 @@ IDENTITY_UAMIS_JSON_MAP=$(echo -n "${IDENTITY_UAMIS_JSON_MAP}" | jq \
 
 }
 
+get_identity_principal_id() {
+  local identity_name="$1"
+  local resource_group="$2"
+  
+  az identity show \
+    --name "${identity_name}" \
+    --resource-group "${resource_group}" \
+    --query principalId \
+    --output tsv
+}
+
+assign_role_to_identity() {
+  local principal_id="$1"
+  local role_name="$2"
+  local scope="$3"
+  
+  echo "Assigning role '${role_name}' to principal ID '${principal_id}' with scope '${scope}'"
+  
+  az role assignment create \
+    --assignee-object-id "${principal_id}" \
+    --assignee-principal-type ServicePrincipal \
+    --role "${role_name}" \
+    --scope "${scope}"
+    
+  echo "Role assignment completed successfully"
+}
+
+create_role_assignments_for_control_plane_identities() {
+  if [[ ! -f "$CONFIG_FILE" ]]; then
+    echo "Warning: CONFIG_FILE not found at ${CONFIG_FILE}. Skipping control plane role assignments."
+    return 0
+  fi
+
+  while read -r operator_name role_definition_resource_id; do
+    if [[ -n "$operator_name" && -n "$role_definition_resource_id" ]]; then
+      # Find the corresponding UAMI name
+      for i in "${!CONTROL_PLANE_OPERATORS_NAMES[@]}"; do
+        if [[ "${CONTROL_PLANE_OPERATORS_NAMES[$i]}" == "$operator_name" ]]; then
+          uami_name="${CONTROL_PLANE_IDENTITIES_UAMIS_NAMES[$i]}"
+          role_definition_id="/subscriptions/${SUBSCRIPTION_ID}${role_definition_resource_id}"
+
+          # Get the principal ID of the managed identity
+          principal_id=$(get_identity_principal_id "${uami_name}" "${CUSTOMER_RG_NAME}")
+          
+          if [[ -n "$principal_id" ]]; then
+            # Assign role at resource group scope
+            assign_role_to_identity "${principal_id}" "${role_definition_id}" "${RESOURCE_GROUP_RESOURCE_ID}"
+          else
+            echo "Error: Could not retrieve principal ID for identity ${uami_name}"
+          fi
+          break
+        fi
+      done
+    fi
+  done < <(yq '.controlPlaneOperatorsIdentities | to_entries[] | .key + " " + .value.roleDefinitions[0].resourceId' "$CONFIG_FILE")
+}
+
+create_role_assignments_for_data_plane_identities() {
+  if [[ ! -f "$CONFIG_FILE" ]]; then
+    echo "Warning: CONFIG_FILE not found at ${CONFIG_FILE}. Skipping data plane role assignments."
```

Comment:
I'm unsure if we should fail silently, or error and stop.
Eventually, if role assignment creation is skipped, it will block cluster creation with the new inflights.